crates/aranya-daemon/src/daemon.rs

```diff
@@ -87,6 +87,9 @@ impl Daemon {
         let mut local_store = self.load_local_keystore().await?;
         let api_sk = self.load_or_gen_api_sk(&mut eng, &mut local_store).await?;
 
+        // Write the daemon's public API key to the UDS directory before creating the socket
+        self.write_api_key_to_uds_dir(&api_sk).await?;
```

Comment:
Let's call this method something else to indicate it's writing the public key and not a private key:
```suggestion
        self.write_api_pk_to_uds_dir(&api_sk).await?;
```